pkg/operator/controller/ingress/controller.go

```diff
@@ -559,6 +563,36 @@ func filterTLS13Ciphers(ciphers []string) []string {
 	return filteredCiphers
 }
 
+// validateHTTPHeaderBufferValues validates the given ingresscontroller's header buffer
+// size configuration, if it specifies one.
+func validateHTTPHeaderBufferValues(ic *operatorv1.IngressController) error {
+	bufSize := int(ic.Spec.HTTPHeaderBuffer.HeaderBufferBytes)
+	maxRewrite := int(ic.Spec.HTTPHeaderBuffer.HeaderBufferMaxRewriteBytes)
+
+	if bufSize == 0 && maxRewrite == 0 {
+		return nil
+	}
+
+	// HeaderBufferBytes and HeaderBufferMaxRewriteBytes are both
+	// optional fields. Substitute the default values used by the
+	// router when either field is empty so that we can ensure that
+	// tune.maxrewrite will never wind up being greater than tune.bufsize
+	// (which would break router reloads).
+	if bufSize == 0 {
+		bufSize = routerDefaultHeaderBufferSize
+	}
+	if maxRewrite == 0 {
+		maxRewrite = routerDefaultHeaderBufferMaxRewriteSize
+	}
+
+	if bufSize < maxRewrite {
```

Comment:
Ah, good catch. I'll change `<` to `=<`.

Reply:
Fixed.

Reply:
I guess using `<=` will catch a lot of errors, but it still allows values like `bufSize = 32768` and `maxRewrite = 32767`, which I imagine would cause all requests to fail, right?

Reply:
> I guess using `<=` will catch a lot of errors, but it still allows values like `bufSize = 32768` and `maxRewrite = 32767`, which I imagine would cause all requests to fail, right?

Yea, that's a good point. Not sure how much more validation we want to do here.... Can we think of a minimum difference between bufSize and maxRewrite that we can safely enforce?